policy/modules/services/container.te

```diff
@@ -0,0 +1,880 @@
+allow container_engine_domain container_file_t:blk_file { manage_blk_file_perms relabel_blk_file_perms };
+allow container_engine_domain container_file_t:chr_file { manage_chr_file_perms relabel_chr_file_perms };
+allow container_engine_domain container_file_t:lnk_file { manage_lnk_file_perms relabel_lnk_file_perms };
+allow container_engine_domain container_file_t:filesystem { getattr relabelfrom relabelto mount unmount remount };
+
+allow container_engine_domain container_ro_file_t:dir { manage_dir_perms relabel_dir_perms };
+allow container_engine_domain container_ro_file_t:file { manage_file_perms relabel_file_perms exec_file_perms };
+allow container_engine_domain container_ro_file_t:blk_file { manage_blk_file_perms relabel_blk_file_perms };
+allow container_engine_domain container_ro_file_t:chr_file { manage_chr_file_perms relabel_chr_file_perms };
+allow container_engine_domain container_ro_file_t:lnk_file { manage_lnk_file_perms relabel_lnk_file_perms };
+
+ifdef(`init_systemd',`
+	# needed by runc, which is also invoked by other engines
+	init_bpf_run(container_engine_domain)
+')
+
+tunable_policy(`container_mounton_non_security',`
+	files_mounton_non_security(container_engine_domain)
+')
+
+tunable_policy(`container_use_nfs',`
+	fs_manage_nfs_dirs(container_engine_domain)
+	fs_manage_nfs_files(container_engine_domain)
+	fs_manage_nfs_named_sockets(container_engine_domain)
+	fs_read_nfs_symlinks(container_engine_domain)
+	fs_mount_nfs(container_engine_domain)
+	fs_unmount_nfs(container_engine_domain)
+	fs_exec_nfs_files(container_engine_domain)
+	kernel_rw_fs_sysctls(container_engine_domain)
+')
+
+tunable_policy(`container_use_samba',`
+	fs_manage_cifs_dirs(container_engine_domain)
+	fs_manage_cifs_files(container_engine_domain)
+	fs_manage_cifs_named_sockets(container_engine_domain)
+	fs_read_cifs_symlinks(container_engine_domain)
+	fs_exec_cifs_files(container_engine_domain)
+')
+
+optional_policy(`
+	# to verify container image signatures
+	gpg_exec(container_engine_domain)
+	gpg_dontaudit_exec_agent(container_engine_domain)
+	gpg_dontaudit_search_user_secrets(container_engine_domain)
+')
+
+optional_policy(`
+	iptables_domtrans(container_engine_domain)
+')
+
+########################################
+#
+# Common system container engine local policy
+#
+
+# system container engines can interact with any container
+ps_process_pattern(container_engine_system_domain, container_domain)
+allow container_engine_system_domain container_domain:process { sigkill signal signull transition };
+allow container_engine_system_domain container_domain:key { create search setattr view };
+
+create_dirs_pattern(container_engine_system_domain, container_config_t, container_config_t)
+files_etc_filetrans(container_engine_system_domain, container_config_t, dir, "cni")
+files_etc_filetrans(container_engine_system_domain, container_config_t, dir, "containers")
+files_etc_filetrans(container_engine_system_domain, container_config_t, dir, "docker")
```

Comment:
Not clear why named transitions are requried.

Reply:
Same as above.